src/main/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfiguration.java

```diff
@@ -1,36 +1,64 @@
 package uk.gov.ida.verifyserviceprovider.configuration;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import io.dropwizard.client.JerseyClientConfiguration;
+
 import uk.gov.ida.saml.metadata.EidasMetadataConfiguration;
+import uk.gov.ida.saml.metadata.TrustStoreConfiguration;
 
 import javax.validation.Valid;
 import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.security.KeyStore;
+
+import static java.util.Optional.ofNullable;
+
+public class EuropeanIdentityConfiguration extends EidasMetadataConfiguration {
 
-public class EuropeanIdentityConfiguration {
+    private final Boolean enabled;
+    private HubEnvironment environment;
 
-    @NotNull
-    @Valid
-    @JsonProperty
-    private String hubConnectorEntityId;
+    @JsonCreator
+    public EuropeanIdentityConfiguration(
+            @JsonProperty("trustAnchorUri") URI trustAnchorUri,
+            @JsonProperty("minRefreshDelay") Long minRefreshDelay,
+            @JsonProperty("maxRefreshDelay") Long maxRefreshDelay,
+            @JsonProperty("trustAnchorMaxRefreshDelay") Long trustAnchorMaxRefreshDelay,
+            @JsonProperty("trustAnchorMinRefreshDelay") Long trustAnchorMinRefreshDelay,
+            @JsonProperty("client") JerseyClientConfiguration client,
+            @JsonProperty("jerseyClientName") String jerseyClientName,
+            @JsonProperty("trustStore") TrustStoreConfiguration trustStore,
+            @JsonProperty("metadataSourceUri") URI metadataSourceUri,
+            @NotNull @Valid @JsonProperty("enabled") Boolean enabled
+    ) {
+        super(trustAnchorUri, minRefreshDelay, maxRefreshDelay, trustAnchorMaxRefreshDelay, trustAnchorMinRefreshDelay, client, jerseyClientName, trustStore, metadataSourceUri);
+        this.enabled = enabled;
+    }
 
-    @NotNull
-    @Valid
-    @JsonProperty
-    private boolean enabled;
+    @JsonIgnore
+    public void setEnvironment(HubEnvironment environment) {
+        this.environment = environment;
+    }
 
-    @Valid
-    @JsonProperty
-    private EidasMetadataConfiguration aggregatedMetadata;
+    @Override
+    public URI getMetadataSourceUri() {
+        return ofNullable(super.getMetadataSourceUri()).orElseGet(() -> environment.getMetadataUri());
+    }
 
-    public String getHubConnectorEntityId() {
-        return hubConnectorEntityId;
+    @Override
+    public KeyStore getTrustStore() {
+        return ofNullable(super.getTrustStore()).orElseGet(() -> environment.getMetadataTrustStore());
     }
 
-    public EidasMetadataConfiguration getAggregatedMetadata() {
-        return aggregatedMetadata;
+    @Override
+    public URI getTrustAnchorUri() {
+        return ofNullable(super.getTrustAnchorUri()).orElseGet(() -> URI.create("http://example.com"));
     }
 
-    public boolean isEnabled() {
+    public Boolean isEnabled() {
```

Comment:
Is there a reason we've changed this into a Boolean object, rather than a boolean primitive? It's only used in one place, and that place coerces it into a primitive anyway; what was wrong with doing the coercion in this class and avoiding exposing a Boolean? Boolean objects open up the unpleasant possibility of nulls.

Reply:
Nicely spotted, this not supposed to be so